From eeaa7b6091c66e78c65ff193ef0ece84d2e740e4 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 13 Dec 2018 10:37:34 +0100
Subject: [PATCH 0966/1200] Revert "arm64: mvebu: cn9x30: Add support for
 CN9X30 demo board"

This reverts commit 01955a206bfc2add0dd179992e55748295f7d780.

Change-Id: Ia67d22286d222b48261c58bebd4c6b167028346a
Reviewed-on: http://vgitil04.il.marvell.com:8080/62081
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1870
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 arch/arm/dts/Makefile                  |   1 -
 arch/arm/dts/marvell-CN9x30-demo-B.dts | 109 -------------------------
 arch/arm/mach-mvebu/armada8k/soc.c     |  40 +++------
 3 files changed, 12 insertions(+), 138 deletions(-)
 delete mode 100644 arch/arm/dts/marvell-CN9x30-demo-B.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 1ef0eeb1ab..5827413eb2 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -106,7 +106,6 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-385-amc.dtb			\
 	armada-3900-vd-A.dtb			\
 	armada-3900-vd-B.dtb			\
-	marvell-CN9x30-demo-B.dtb		\
 	armada-7020-amc.dtb			\
 	armada-7040-pcac.dtb	                \
 	armada-7040-db.dtb			\
diff --git a/arch/arm/dts/marvell-CN9x30-demo-B.dts b/arch/arm/dts/marvell-CN9x30-demo-B.dts
deleted file mode 100644
index 4a06c75eaf..0000000000
--- a/arch/arm/dts/marvell-CN9x30-demo-B.dts
+++ /dev/null
@@ -1,109 +0,0 @@
-/*
- * Copyright (C) 2018 Marvell International Ltd.
- *
- * SPDX-License-Identifier:    GPL-2.0
- * https://spdx.org/licenses
- */
-
-#include "armada-3900-db.dtsi"
-
-/ {
-	model = "Model: Marvell CN9x30 demo board (CP NOR) setup(B)";
-	compatible = "marvell,cn9x30-demo", "marvell,armada3900-vd",
-		     "marvell,armada7040", "marvell,armada-ap806-quad",
-		     "marvell,armada-ap806";
-
-	aliases {
-		i2c1 = &ap_i2c0;
-	};
-};
-
-/*
- * AP related configuration
- */
-&ap_i2c0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&ap_i2c0_pins>;
-	status = "okay";
-};
-
-/*
- * CP related configuration
- */
-&cp0_pinctl {
-		/* MPP Bus:
-		 *   NAND	[13,15-27]
-		 *   SMI	[32,34]
-		 *   XSMI	[35-36]
-		 *   I2C0	[37-38]
-		 *   USB	[44-45]
-		 *   UART1	[46-47,49,58]
-		 *   IHB	[56-57]
-		 *   UART0	[59-62]
-		 */
-		/*   0   1   2   3   4   5   6   7   8   9 */
-	pin-func = < 0   0   0   0   0   0   0   0   0   0
-		     0   0   0   2   0   1   1   1   1   1
-		     1   1   1   1   1   1   1   1   0   0
-		     0   0   7   0   7   7   7   2   2   0
-		     0   0   0   0   0   0   7   7   0   7
-		     0   0   0   0   0   0   0   0   7   7
-		     7   7   7>;
-};
-
-&cp0_nand {
-	status = "okay";
-};
-
-&cp0_comphy {
-
-	phy0 {
-		phy-type = <COMPHY_TYPE_PEX0>;
-		/delete-property/phy-speed;
-	};
-
-	phy1 {
-		phy-type = <COMPHY_TYPE_USB3_HOST0>;
-	};
-
-	phy2 {
-		phy-type = <COMPHY_TYPE_SFI0>;
-		phy-speed = <COMPHY_SPEED_5_15625G>;
-	};
-
-	phy3 {
-		phy-type = <COMPHY_TYPE_UNCONNECTED>;
-		/delete-property/phy-speed;
-	};
-
-	phy4 {
-		phy-type = <COMPHY_TYPE_SGMII1>;
-		phy-speed = <COMPHY_SPEED_1_25G>;
-	};
-};
-
-&cp0_xmdio {
-	status = "okay";
-	phy0: ethernet-phy@0 {
-		reg = <0>;
-	};
-	phy1: ethernet-phy@1 {
-		reg = <1>;
-	};
-};
-
-&cp0_ethernet {
-	status = "okay";
-};
-
-&cp0_eth0 {
-	status = "okay";
-	phy-mode = "sfi";
-	phy = <&phy0>;
-};
-
-&cp0_eth1 {
-	status = "okay";
-	phy-mode = "sgmii";
-	phy = <&phy1>;
-};
diff --git a/arch/arm/mach-mvebu/armada8k/soc.c b/arch/arm/mach-mvebu/armada8k/soc.c
index 6d83bca97b..5473f1981b 100644
--- a/arch/arm/mach-mvebu/armada8k/soc.c
+++ b/arch/arm/mach-mvebu/armada8k/soc.c
@@ -5,11 +5,10 @@
  * https://spdx.org/licenses
  */
 
+#include <common.h>
 #include <asm/arch-armada8k/cache_llc.h>
 #include <asm/io.h>
 #include <asm/arch/soc.h>
-#include <common.h>
-#include <dm/device.h>
 #include <mvebu/mvebu_chip_sar.h>
 
 #define CP_DEV_ID_STATUS_REG		(MVEBU_REGISTER(0x2400240))
@@ -64,15 +63,6 @@ static struct soc_info soc_info_table[] = {
 	{ {0x8040, 1}, "Armada8040-A1", {0x806, 1}, {0x110, 1}, 1, 2 },
 	{ {0x8040, 2}, "Armada8040-A2", {0x806, 1}, {0x110, 2}, 1, 2 },
 	{ {0x8045, 0}, "Armada8040-B0", {0x806, 2}, {0x115, 0}, 1, 2 },
-	/*****************WARNING*********************
-	 * Keep the below entry as last in this array!
-	 * This entry is for SoC demo purpose only!
-	 * There is no way to distinguish between A3900
-	 * and cn9x30 on demo board until the device ID
-	 * is changed in the SoC HW.
-	 *****************WARNING*********************
-	 */
-	{ {0x6025, 0}, "CN9X30_DEMO", {0x807, 1}, {0x115, 0}, 1, 1, 0},
 };
 
 static int get_soc_type_rev(u32 *type, u32 *rev)
@@ -119,23 +109,17 @@ static int get_soc_table_index(u32 *index)
 	get_soc_type_rev(&soc_type, &rev);
 	get_ap_soc_type(&ap_type);
 
-	if (!of_machine_is_compatible("marvell,cn9x30-demo")) {
-		for (i = 0; i < ARRAY_SIZE(soc_info_table) && ret != 0; i++) {
-			if ((soc_type != soc_info_table[i].soc.module_type) ||
-			    (rev != soc_info_table[i].soc.module_rev) ||
-			    ap_type != soc_info_table[i].ap.module_type)
-				continue;
-
-			if (!get_soc_sub_rev(&sub_rev) &&
-			    (sub_rev != soc_info_table[i].sub_rev))
-				continue;
-
-			*index = i;
-			ret = 0;
-		}
-	} else {
-		/* For demo board just use the last table entry */
-		*index = ARRAY_SIZE(soc_info_table) - 1;
+	for (i = 0; i < ARRAY_SIZE(soc_info_table) && ret != 0; i++) {
+		if ((soc_type != soc_info_table[i].soc.module_type) ||
+		    (rev != soc_info_table[i].soc.module_rev) ||
+		    ap_type != soc_info_table[i].ap.module_type)
+			continue;
+
+		if (!get_soc_sub_rev(&sub_rev) &&
+		    (sub_rev != soc_info_table[i].sub_rev))
+			continue;
+
+		*index = i;
 		ret = 0;
 	}
 
-- 
2.22.0

